csi/controller_server.go

```diff
@@ -269,7 +269,7 @@ func (cs *ControllerServer) ControllerPublishVolume(ctx context.Context, req *cs
 	}
 
 	// the volume is already attached make sure it's to the same node as this
-	if existVol.State == string(types.VolumeStateAttached) {
+	if existVol.State == string(types.VolumeStateAttached) && existVol.Controllers[0].Endpoint == "" {
```

Comment:
`Endpoint == ""`??

Also, why didn't we check if the frontend is disabled or not, or if frontend is blockdev or not? Just like we do in the node publish volume.

Reply:
Will fix it. Thanks.